src/js/libcs/Render2D.js

```diff
@@ -619,6 +619,36 @@ Render2D.drawline = function(homog) {
     }
 };
 
+// draws a segment consisting of 2 rays
+Render2D.drawRaySegment = function(A, B, C, sflip) {
+    var ptA = eval_helper.extractPoint(A);
+    var ptB = eval_helper.extractPoint(B);
```

Comment:
`B` might actually be at infinity, turning `ptB` into numeric garbage (Infinities, NaN and the likes). For certain scenarios it's even likely to have the midpoint mapped to infinity, so this is not just a corner case with probability zero.